### DISCUSSION TITLE
Obtain the Schmid Factor per grain

### DISCUSSION THREAD

**Discussion by Xavier-Ojeda** [MechMicroMan/DefDAP#115] [General]
Hello! I was wondering if there is way to calculate the Schmid Factor per each grain (the exact/approximate value) and export that information as a list or array. The plot average Schmid Factor function is not what I need to assigned a factor to each grain in my data set.

**Comment by mikesmic** [comment]
Hi @Xavier-Ojeda for an EBSD map there are functions to calculate the Schmid factors for each grain (`grain.calcAverageSchmidFactors`) and for all grains in the map (`map.calcAverageGrainSchmidFactors`). You pass the load vector and can optionally pass a nested list of slip systems but by default will use a set of slip systems based on the crystal structure of the phase and these are grouped by the slip plane (grouping can be changed if needed). The result is stored in each grain object (`grain.averageSchmidFactors`) as a nested list of lists grouped by slip plane by default. If you are working with a HRDIC map, you would calculate Schmid factor for the linked EBSD map and then the results can be accessed through each DIC grain via the linked EBSD grain (`grain.ebsdGrain.averageSchmidFactors`). There are also some helper functions for plotting values per grain, have a look through `base.py` for these writing your own code.
